easybuild/easyconfigs/r/ReFrame/ReFrame-3.0.eb

```diff
@@ -0,0 +1,93 @@
+easyblock = 'PythonPackage'
+
+name = 'ReFrame'
+version = '3.0'
+
+homepage = 'https://github.com/eth-cscs/reframe'
+description = '''ReFrame is a framework for writing regression tests for HPC systems.'''
+
+toolchain = SYSTEM
+
+
+sources = [{
+    'filename': 'v%(version)s.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/eth-cscs',
+        'repo_name': 'reframe',
+        'tag': 'v%(version)s'
+    }
+}]
+
+checksums = ['8ecf8eeed11ad1d77d351c06d3c21775d62c16e4540bf66da8da0ed25b2004b2']
+
+# ReFrame requires Python 3.6, may only be available via 'python3'
+req_py_majver = 3
+req_py_minver = 6
+
+download_dep_fail = True
+use_pip = False
+
+exts_defaultclass = 'PythonPackage'
+
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+    'download_dep_fail': True,
+    'use_pip': False
+}
+
+exts_list = [
+    ('pip', '20.1.1', {
+        'use_pip': False,
+        'checksums': ['27f8dc29387dd83249e06e681ce087e6061826582198a425085e0bf4c1cf3a55'],
+    }),
+    ('setuptools', '47.1.1', {
+        'source_tmpl': 'setuptools-47.1.1.zip',
+        'checksums': ['145fa62b9d7bb544fce16e9b5a9bf4ab2032d2f758b7cd674af09a92736aff74'],
+    }),
+    ('zipp', '3.1.0', {
+        'use_pip': True,
+        'source_tmpl': 'zipp-3.1.0-py3-none-any.whl',
+        'unpack_sources': False,
+        'checksums': ['aa36550ff0c0b7ef7fa639055d797116ee891440eac1a56f378e2d3179e0320b'],
+    }),
+    ('attrs', '19.3.0', {
+        'modulename': 'attr',
+        'checksums': ['f7b7ce16570fe9965acd6d30101a28f62fb4a7f9e926b3bbc9b61f8b04247e72'],
+    }),
+    ('importlib_metadata', '1.6.0', {
+        'checksums': ['34513a8a0c4962bc66d35b359558fd8a5e10cd472d37aec5f66858addef32c1e'],
+    }),
+    ('pygelf', '0.3.6', {
+        'checksums': ['3e5bc59e3b5a754556a76ff2c69fcf2003218ad7b5ff8417482fa1f6a7eba5f9'],
+    }),
+    ('pyrsistent', '0.16.0', {
+        'checksums': ['28669905fe725965daa16184933676547c5bb40a5153055a8dee2a4bd7933ad3'],
+    }),
+    ('jsonschema', '3.2.0', {
+        'checksums': ['c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a'],
+    }),
+]
+
+postinstallcmds = [
+    'cd %(builddir)s/reframe && cp -r tutorial %(installdir)s && '
+    'python3 -m venv docs_env && source docs_env/bin/activate && '
+    'pip install --no-cache-dir -r docs/requirements.txt && make -C docs && deactivate && '
```

Comment:
We should try avoiding `pip install` here, since the Python packages being installed here are not reproducible (things will change as new versions are released, and this is bound to break at some point).

Is there a "release source tarball" available that comes with the docs pre-built?

Reply:
The only reason for this is to create the man pages. No there is no release tarball containing the man pages.